src/Client.js

```diff
@@ -276,6 +285,14 @@ Client.prototype.request = function (options, params, callback) {
     // Add all the given parameters to the querystring.
     req = req.query(queryParams);
 
+    if (isFunction(reqCustomizer)) {
+      reqCustomizer(req, params);
+    }
+
+    if (isFunction(params._requestCustomizer)) {
+      params._requestCustomizer(req, params);
```

Comment:
I think I would call this a "middleware" instead of a "customizer".

When a middleware is passed to a method (e.g. `create()`), shouldn't it replace the global middleware (the one passed to the Client instance) ?

Let's say you want a middleware to be applied to all requests except for the `getAll()` requests. In the current implementation, you have no way to disable or override the middleware function, because the one in `this.options.request.customizer` will always be executed.

I'd propose the following change:
```javascript
// ...
req = req.query(queryParams);

// If a middleware was passed to the request() method, apply it.
if (isFunction(params._middleware)) {
  params._middleware(req, params);
} 
// If middewares were disabled (params._middleware === false), ignore the middleware
// in this.options.request.middleware, otherwise, apply it.
else if ((params._middleware !== false) && isFunction(this.options.request.middleware)) {
  this.options.request.middleware(req, params);
}
```

That way you could define a global middleware when creating the Client instance, replace it when calling a method, or simply disable it and not apply any middleware to a given method call.